### DISCUSSION TITLE
Local Integration

### DISCUSSION THREAD

**Discussion by explosivo22** [explosivo22/rinnaicontrolr-ha#54] [Polls]
We have a new discovery that we could potentially be able to connect to the water heater locally just by knowing the host address.  We would be able to have the same sensors and services as before with just a few attributes missing.

**Comment by wrmacj** [comment]
I have voted Local only but wonder Is there a drawback to that?  Would it be a mater of installing the integration and then just possibly entering the ipaddress of the device so that the system could connect?

**Reply by explosivo22** [reply]
I think that is exactly how it would have to work. Install the integration, and then it would prompt for the IP address. The only thing that would be lost that way that I have found so far is the device name. You would have to manually name it in HA, but that isn't a big issue to me.

**Reply by wrmacj** [reply]
Perfect and simple.  Maybe populate a default name so all the user has to do is enter the ip address?

**Comment by YseanY** [comment]
Anything new regarding local integration?

**Reply by explosivo22** [reply]
I am working on getting a test branch together with the rewrite for those who want to try it can

**Reply by YseanY** [reply]
Awesome! Looking forward to it!

**Reply by YseanY** [reply]
> I am working on getting a test branch together with the rewrite for those who want to try it can

Any progress/ETA on a local integration?

**Reply by explosivo22** [reply]
Just created a Pre-Release for this!

**Reply by YseanY** [reply]
> Just created a Pre-Release for this!

Thank you! 🙏🏻 

but excuse my noob question, how do I switch over to the local integration in HA?

**Reply by explosivo22** [reply]
You can get there with the following steps:

1. Settings
2. Devices and Services
3. HACS
4. Click on services
5. Click on `Rinnai Control-R Water Heater`
6. Under Diagnostic, toggle Pre-release

If you don't have the option, you will need to enable, but not toggle on, the Pre-release option on the HACS service and then you can go back to the Rinnai service and will be able to enable and toggle on the option for Rinnai.

Alternatively, you can just go into HACS and find `Rinnai Control-R Water Heater` and click the three dots and choose `Redownload`.  Under the **`Need a different version?`**, you can select 2.1.1 and it will get installed as well, but you won't be notified if any newer versions are released.

**Reply by YseanY** [reply]
> You can get there with the following steps:
> 
> 1. Settings
> 2. Devices and Services
> 3. HACS
> 4. Click on services
> 5. Click on `Rinnai Control-R Water Heater`
> 6. Under Diagnostic, toggle Pre-release
> 
> If you don't have the option, you will need to enable, but not toggle on, the Pre-release option on the HACS service and then you can go back to the Rinnai service and will be able to enable and toggle on the option for Rinnai.
> 
> Alternatively, you can just go into HACS and find `Rinnai Control-R Water Heater` and click the three dots and choose `Redownload`. Under the **`Need a different version?`**, you can select 2.1.1 and it will get installed as well, but you won't be notified if any newer versions are released.

Awesome! Just updated and seems to be working perfectly

**Comment by srench01** [comment]
I am using the local integration (as of today).  is there a way for the integration to expose the Diagnostic codes?

My heater is outdoor - and sits about 10' high.  For me to see the display screen, I have to get a ladder.  Today when I used the Control R app (which I hadn't looked at in a long time) I noticed that I had "errors".  The errors in the app only said "null".  When I went outside on the ladder - the display was flashing the desired temp (120) as well as the code "55" (guessing this is supposed to be "SS" for SERVICE SOON.  

I was wondering if that code "SS" (or 55 ?) could be exposed so that I could create an automation when that is triggered to do something.  Flash lights, send an email, whatever - so that then I knew I need to go take care of the unit.  

My unit is an outdoor RE199EP

I also found a Tech Sheet that has the Diagnostic Codes (right side of first page) among many other parameters and settings.

(and I was in fact due for the annual flushing of the system)

[100000735 VE Non-Pump Outdoor Tech Sheet.pdf](https://github.com/user-attachments/files/19147107/100000735.VE.Non-Pump.Outdoor.Tech.Sheet.pdf)
